tests/model_registry/async_job/utils.py

```diff
@@ -0,0 +1,89 @@
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.job import Job
+from ocp_resources.pod import Pod
+
+from tests.model_registry.async_job.constants import (
+    ASYNC_JOB_ANNOTATIONS,
+    ASYNC_JOB_LABELS,
+    MODEL_SYNC_CONFIG,
+    VOLUME_MOUNTS,
+)
+
+
+def get_job_pod(admin_client: DynamicClient, job: Job) -> Pod:
+    """Get the Pod created by a Job"""
+    pods = list(
+        Pod.get(
+            dyn_client=admin_client,
+            namespace=job.namespace,
+            label_selector=f"job-name={job.name}",
+        )
+    )
+    assert len(pods) == 1, f"Expected 1 pod for job {job.name}, found {len(pods)}"
+    return pods[0]
+
+
+def validate_job_labels_and_annotations(job: Job) -> None:
+    """Validate Job has all required labels and annotations"""
+    job_labels = job.instance.metadata.labels or {}
+    job_annotations = job.instance.metadata.annotations or {}
+
+    # Validate required labels
+    for key, expected_value in ASYNC_JOB_LABELS.items():
+        actual_value = job_labels.get(key)
+        assert actual_value == expected_value, f"Label {key}: expected {expected_value}, got {actual_value}"
+
+    # Validate required annotations
+    for key, expected_value in ASYNC_JOB_ANNOTATIONS.items():
+        actual_value = job_annotations.get(key)
+        assert actual_value == expected_value, f"Annotation {key}: expected {expected_value}, got {actual_value}"
+
+
+def validate_job_pod_template_labels(job: Job) -> None:
+    """Validate Job pod template has required model sync labels"""
+    pod_template_labels = job.instance.spec.template.metadata.labels or {}
```

Comment:
All the validations would require, api calls. Do you want to send the job dict instead of job object?

Reply:
Wouldn't this require a single api call here and then the labels are stored in pod_template_labels, and we're just fetching them by key a few lines down?